src/H5FDsubfiling/H5FDsubfiling.c

```diff
@@ -1657,71 +1596,44 @@ H5FD__subfiling_write(H5FD_t *_file, H5FD_mem_t type, hid_t H5_ATTR_UNUSED dxpl_
- *
  *              Failure:    FAIL
- *                          The contents of supplied buffers are undefined.
- *
- * Notes:       Thus function doesn't actually implement vector read.
- *              Instead, it converts the vector read call into a series
- *              of scalar read calls.  Fix this when time permits.
- *
- *              Also, it didn't support the sizes and types optimization.
- *              I implemented a version of this which is more generous
- *              than that currently defined in the RFC.  This is good
- *              enough for now, but the final version should follow
- *              the RFC.
- *                                                    JRM -- 10/5/21
  *
  *-------------------------------------------------------------------------
  */
 static herr_t
 H5FD__subfiling_read_vector(H5FD_t *_file, hid_t dxpl_id, uint32_t count, H5FD_mem_t types[], haddr_t addrs[],
                             size_t sizes[], void *bufs[] /* out */)
 {
-    H5FD_subfiling_t *file_ptr  = (H5FD_subfiling_t *)_file;
+    H5FD_subfiling_t *file      = (H5FD_subfiling_t *)_file;
     herr_t            ret_value = SUCCEED;
 
-    assert(file_ptr);
-    assert(file_ptr->pub.driver_id == H5FD_SUBFILING);
-    assert((types) || (count == 0));
-    assert((addrs) || (count == 0));
-    assert((sizes) || (count == 0));
-    assert((bufs) || (count == 0));
+    FUNC_ENTER_PACKAGE
 
-    /*
-     * Verify that the first elements of the sizes and
-     * types arrays are valid.
-     */
-    assert((count == 0) || (sizes[0] != 0));
-    assert((count == 0) || (types[0] != H5FD_MEM_NOLIST));
+    assert(file);
+    assert(count == 0 || types);
+    assert(count == 0 || addrs);
+    assert(count == 0 || sizes);
+    assert(count == 0 || bufs);
+
+    /* Verify that the first elements of the sizes and types arrays are valid */
+    assert(count == 0 || sizes[0] != 0);
+    assert(count == 0 || types[0] != H5FD_MEM_NOLIST);
 
     /* Get the default dataset transfer property list if the user didn't provide one */
-    if (H5P_DEFAULT == dxpl_id) {
+    if (H5P_DEFAULT == dxpl_id)
         dxpl_id = H5P_DATASET_XFER_DEFAULT;
-    }
-    else {
-        if (true != H5P_isa_class(dxpl_id, H5P_DATASET_XFER))
-            H5_SUBFILING_GOTO_ERROR(H5E_ARGS, H5E_BADTYPE, FAIL, "not a data transfer property list");
-    }
+    else if (true != H5P_isa_class(dxpl_id, H5P_DATASET_XFER))
+        HGOTO_ERROR(H5E_ARGS, H5E_BADTYPE, FAIL, "not a data transfer property list");
 
     /* Set DXPL for operation */
     H5CX_set_dxpl(dxpl_id);
 
-    if (file_ptr->fa.require_ioc) {
-        if (H5FD__subfiling_io_helper(file_ptr, (size_t)count, types, addrs, sizes,
-                                      (H5_flexible_const_ptr_t *)bufs, IO_TYPE_READ) < 0)
-            H5_SUBFILING_GOTO_ERROR(H5E_IO, H5E_READERROR, FAIL, "couldn't read data");
-    }
-    else {
-        if (H5FD_read_vector(_file, count, types, addrs, sizes, bufs) < 0)
```

Comment:
This called was removed here, but left for `H5FD__subfiling_read_vector`. While I'm not really in favor of removing things functionality-wise unless there is a demonstrated need for it, the two should be consistent either way.